exec/java-exec/src/main/java/org/apache/drill/exec/work/foreman/QueryManager.java

```diff
@@ -379,6 +379,12 @@ private QueryProfile getQueryProfile(UserException ex) {
       profileBuilder.setQuery(queryText);
     }
 
+    final int autoLimitRowCount = foreman.getQueryContext().getOptions().getOption(ExecConstants.QUERY_MAX_ROWS).num_val.intValue();
```

Comment:
Please remove `final` modifier here and in other places where it is not required.